docs/orchestrating-elastic-stack-applications/logstash.asciidoc

```diff
@@ -837,15 +834,501 @@ spec:
+== {ls} plugins
+
+The power of {ls} is in the plugins--{logstash-ref}/input-plugins.html[inputs], {logstash-ref}/output-plugins.html[outputs], {logstash-ref}/filter-plugins.html[filters,] and {logstash-ref}/codec-plugins.html[codecs].
+
+In {ls} on ECK, you can use the same plugins that you use for other {ls} instances--including Elastic-supported, community-supported, and custom plugins.
+However, you may have other factors to consider, such as how you configure your {k8s} resources, how you specify additional resources, and how you scale your {ls} installation.  
+
+In this section, we'll cover: 
+
+* <<{p}-plugin-resources,Providing additional resources for plugins (read-only and writable storage)>>
+* <<{p}-logstash-working-with-plugins-scaling>>
+* <<{p}-logstash-working-with-plugin-considerations>>
+* <<{p}-logstash-working-with-custom-plugins>>
+
+[id="{p}-plugin-resources"]
+=== Providing additional resources for plugins
+
+The plugins in your pipeline can impact how you can configure your {k8s} resources, including the need to specify additional resources in your manifest.
+The most common resources you need to allow for are:
+
+* Read-only assets, such as private keys, translate dictionaries, or JDBC drivers
+* <<{p}-logstash-working-with-plugins-writable>> to save application state
+
+[id="{p}-logstash-working-with-plugins-ro"]
+==== Read-only assets
+
+Many plugins require or allow read-only assets in order to work correctly.
+These may be ConfigMaps or Secrets files that have a 1 MiB limit, or larger assets such as JDBC drivers, that need to be stored in a PersistentVolume.
+
+[id="{p}-logstash-working-with-plugins-small-ro"]
+===== ConfigMaps and Secrets (1 MiB max)
+
+Each instance of a `ConfigMap` or `Secret` has a https://kubernetes.io/docs/concepts/configuration/configmap/#:~:text=The%20data%20stored%20in%20a,separate%20database%20or%20file%20service[maximum size] of 1 MiB (mebibyte). 
+For larger read-only assets, check out <<{p}-logstash-working-with-plugins-large-ro>>.
+
+In the plugin documentation, look for configurations that call for a `path` or an `array` of `paths`. 
+
+**Sensitive assets, such as private keys**
+
+Some plugins need access to private keys or certificates in order to access an external resource.
+Make the keys or certificates available to the {ls} resource in your manifest.
+
+TIP: These settings are typically identified by an `ssl_` prefix, such as `ssl_key`, `ssl_keystore_path`, `ssl_certificate`, for example.
+
+To use these in your manifest, create a Secret representing the asset, a Volume in your `podTemplate.spec` containing that Secret, and then mount that Volume with a VolumeMount in the `podTemplateSpec.container` section of your {ls} resource.
+
+First, create your secrets.
+
+[source,bash]
+----
+kubectl create secret generic logstash-crt --from-file=logstash.crt
+kubectl create secret generic logstash-key --from-file=logstash.key
+----
+
+Then, create your Logstash CRD.
```

Comment:
```suggestion
Then, create your Logstash resource.
```
or `object` or `manifest`. CRD is the custom resource definition that extends the k8s API and makes it possible to create Logstash objects in the first place, so to speak the defintion of this "definition".